pkg/proc/eval.go

```diff
@@ -28,6 +29,13 @@ func (scope *EvalScope) EvalExpression(expr string, cfg LoadConfig) (*Variable,
 		defer close(scope.callCtx.continueRequest)
 	}
 	t, err := parser.ParseExpr(expr)
+	if el, isScannerErr := err.(scanner.ErrorList); scope.callCtx != nil && isScannerErr && el[0].Msg == "expected '==', found '='" {
```

Comment:
nit: we should put this logic into a function that's a bit more descriptive, such as `isAssignment`.